pkg/api/validation/validation.go

```diff
@@ -1603,6 +1603,21 @@ func ValidateSecret(secret *api.Secret) errs.ValidationErrorList {
 		if err := json.Unmarshal(dockercfgBytes, &map[string]interface{}{}); err != nil {
 			allErrs = append(allErrs, errs.NewFieldInvalid(fmt.Sprintf("data[%s]", api.DockerConfigKey), "<secret contents redacted>", err.Error()))
 		}
+	case api.SecretTypeBasicAuth:
```

Comment:
For ssh-privatekey, "privatekey" is a common identifier used by SSH config,
that's the only reason.

On Mon, Oct 19, 2015 at 3:58 PM, Brian Grant <notifications@github.com>
wrote:

> In pkg/api/validation/validation.go
> <https://github.com/kubernetes/kubernetes/pull/13853#discussion_r42417980>
> :
>
> > @@ -1603,6 +1603,21 @@ func ValidateSecret(secret *api.Secret) errs.ValidationErrorList {
> >  		if err := json.Unmarshal(dockercfgBytes, &map[string]interface{}{}); err != nil {
> >  			allErrs = append(allErrs, errs.NewFieldInvalid(fmt.Sprintf("data[%s]", api.DockerConfigKey), "<secret contents redacted>", err.Error()))
> >  		}
> > +	case api.SecretTypeBasicAuth:
>
> If Kubernetes itself is not consuming these secret types in any way, then
> we shouldn't validate them in the apiserver.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/13853/files#r42417980>.
>